### PR TITLE
PWG/EMCAL: Added Cell energy and time compression component

### PR DESCRIPTION
**Pull request by jokonig**
- Component to try settings for a compression algorithm used in
O2/DataFormats/EMCAL/...Cell.cxx
- Cell energy/time gets multiplied with a certain resolution factor and
then gets transformed in an integer
- All values for resolution etc. can be set in the yaml file

### CODE REVIEW

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  void SetEnergy(Double_t e);
+  Double_t GetEnergy() const;
+  void SetTime(Double_t t);
+  Double_t GetTime() const;
+
+  TH1F                  *fCellEnergyCompressionLoss;            //!<! difference between energy with no compression and with compression
+  TH1F                  *fCellTimeCompressionLoss;              //!<! difference between cell time with no compression and with compression
+
+  Bool_t                 fDoEnergyCompression;                  ///< switch to turn on energy Compression
+  int16_t                fEnergy;                               ///< current energy
```
Comment (mfasDa) [inline PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h:42]: This needs the header <cstdint> to be included, but not sure CINT can use it, better use short here and convert to int16_t in the cxx file, or exclude the class from the useless ROOT5 builds. There might be a different trick using typedefs, like

```c++
#if ROOT_VERSION_CODE < ROOT_VERSION(5,99,0)
typedef eint16_t short;
#else
#include <cstdint>
typedef eint16_t int16_t;
#endif
...
eint16_t fEnergy;
...
```

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  Double_t               fTruncationMaxE;                       ///< Maximum cell energy
+  Double_t               fEnergyCompression;                    ///< Energy Compression
+  Double_t               fCompressionBias;                      ///< bias added to energy before converting it to int16_t
+  Bool_t                 fDoTimeCompression;                    ///< switch to turn on time Compression
+  int16_t                fTime;                                 ///< current time
```
Comment (mfasDa) [inline PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h:47]: Same as above.

**Review by mfasDa — CHANGES_REQUESTED** [review]
Some change is needed for the useless compatibility to ROOT5 only there thanks to nonsense decisions of the ALICE management.

**Comment by mfasDa** [comment]
+test

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h (hunk 1 above)
Comment (jokonig) [reply PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellEnergyCompression.h]: I now just went with Short_t.
Is there any disadvantages the shorts could have in our use case? (I could not think if one)

**Comment by mfasDa** [comment]
+1

**Pull request merged**